forbidigo/forbidigo.go

```diff
@@ -163,44 +195,198 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
+
+	// If we are lucky, the entire selector expression has a known
+	// type. We don't care about the value.
+	selectorText := v.textFor(selector)
+	if typeAndValue, ok := v.runConfig.TypesInfo.Types[selector]; ok {
+		prefix := typeAndValue.Type.String()
+		matchText = finalizeSelectorText(prefix, field)
+		pkgText = pkgFromType(typeAndValue.Type)
+		v.runConfig.DebugLog("%s: selector %q with known type %q: %q -> %q, package %q", location, selectorText, prefix, srcText, matchText, pkgText)
+		return
+	} else {
+		// Some expressions need special treatment.
+		switch selector := selector.(type) {
+		case *ast.Ident:
+			object, ok := v.runConfig.TypesInfo.Uses[selector]
+			if !ok {
+				// No information about the identifier. Should
+				// not happen, but perhaps there were compile
+				// errors?
+				return
+			}
+			switch object := object.(type) {
+			case *types.PkgName:
+				pkgText = object.Imported().Path()
+				matchText = finalizeSelectorText(pkgText, field)
+				v.runConfig.DebugLog("%s: selector %q is package: %q -> %q, package %q", location, selectorText, srcText, matchText, pkgText)
+				return
+			case *types.Var:
+				prefix := object.Type().String()
+				matchText = finalizeSelectorText(prefix, field)
+				pkgText = pkgFromType(object.Type())
+				v.runConfig.DebugLog("%s: selector %q is variable of type %q: %q -> %q, package %q", location, selectorText, prefix, srcText, matchText, pkgText)
+				return
+			default:
+				// Something else?
+				v.runConfig.DebugLog("%s: selector %q is identifier with unsupported type %T", location, selectorText, object)
+				return
+			}
+		default:
+			v.runConfig.DebugLog("%s: selector %q of unsupported type %T", location, selectorText, selector)
+			return
+		}
+	}
+}
+
+// importPathBase returns the last element of a package import path. It ignores
+// a version suffix. The path string must be valid.
+func importPathBase(p string) string {
+	// Keep only the path, not the version. We don't need to check
+	// for success, this cannot fail.
+	p, _, _ = module.SplitPathVersion(p)
+	// Keep the last path element.
+	p = path.Base(p)
+	return p
+}
+
+// pkgFromType tries to deduce the package name from a type.
+func pkgFromType(t types.Type) string {
+	// "Parsing" the string representation isn't ideal, but
+	// the generic Type interface doesn't have anything better.
+	typeString := t.String()
+
+	// The type name, in contrast to path elements, may not contain a dot,
+	// so use the last dot as separator.
+	dot := strings.LastIndex(typeString, ".")
+	if dot < 0 {
+		return ""
+	}
+	pkgText := typeString[:dot]
+
+	// Ignore whether it is a pointer.
+	pkgText = strings.TrimLeft(pkgText, "*")
+
+	// The type is not necessarily <package>.<identifier>. It could also
+	// be a local, unexported type. Sanity check that.
+	if err := module.CheckImportPath(pkgText); err != nil {
+		return ""
+	}
+	return pkgText
+}
+
+// composeSelectorText takes the type prefix, simplifies it and combines it
```

Comment:
The comment doesn't match the method name.  I'm also finding it a bit tricky to understand the role of this method.  A few more examples like you had done for updateText might be helpful here.